### ISSUE TITLE
Loading LORA weights with `load_lora_weights` do not respect `HF_HUB_OFFLINE` / `local_file_only`

### ISSUE THREAD

**Issue by Wauplin** [huggingface/diffusers#6089]
First reported by @rvorias in https://github.com/huggingface/huggingface_hub/issues/1893#issuecomment-1845134191.

**TL;DR:** Loading lora weights on a pipeline makes a call to the Hub even when `HF_HUB_OFFLINE=1` is set as environment variable (or `local_file_only=True` is passed as parameter), which is not expected.

This seems to be due to the `_best_guess_weight_name` method that try to guess which file to load when the LoRA filename is not provided. [It makes a call to `model_info`](https://github.com/huggingface/diffusers/blob/6bf1ca2c799f3f973251854ea3c379a26f216f36/src/diffusers/loaders/lora.py#L307) to check which files exist on the Hub repo.

```py
pipeline.load_lora_weights(
  "latent-consistency/lcm-lora-sdxl",
  adapter_name="lcm",
  cache_dir=cache_dir,
  local_files_only=True,
)
```

**Suggestions:**
- raise an error if offline mode => "hey you should pass a filename"
- check local cache if offline mode => try to guess based on that locally saved files?

**Comment by sayakpaul** [comment]
Maybe it's better to be explicit here so, I would prefer the first option.

**Comment by Wauplin** [comment]
Agree on raising an explicit error while the problem is not solved but it would be good to be able to load lora weights while being offline, no? I think it just requires to list the local files (`os.listdir(snapshot_path)`) instead of listing files from the Hub. 

And I'll open an issue in `huggingface_hub` to raise explicit error in all method of `HfApi` if `HF_HUB_OFFLINE=1` (currently only the case for download methods).

**Comment by sayakpaul** [comment]
>  I think it just requires to list the local files (os.listdir(snapshot_path)) instead of listing files from the Hub.

Hmm, so when `local_files_only=True` we do `os.listdir(...)`, yeah? Yeah, that makes sense too.

**Comment by Wauplin** [comment]
I created https://github.com/huggingface/huggingface_hub/pull/1899 in `huggingface_hub` to raise an explicit error if someone tries to call the Hub with `HF_HUB_OFFLINE=1`. It will make it more obvious to the user what happens.
`diffusers` will have to deal with `OfflineModeIsEnabled` exception the same way it deals with `local_files_only=True`. Instead of catching the error, another possibility is to look at the `huggingface_hub.constant.HF_HUB_OFFLINE` constant directly (i.e. have a condition `if local_files_only or constants.HF_HUB_OFFLINE: ...`.

**Comment by sayakpaul** [comment]
@Wauplin thanks for your inputs. I will take a closer look soon and open a PR asking for your review.
